### DISCUSSION TITLE
V2: Down with `delay`?

### DISCUSSION THREAD

**Discussion by jon-sully** [excid3/noticed#361] [General]
Looks like the `delay` option is no longer supported for delivery methods? I know in V1 it wasn't supported for `deliver_by :database` because it didn't make sense, but I think now that the Event is split from the Notification we could actually support `delay` by just doing `@event.created_at + delay` as the `wait:` param when kicking off the [delivery jobs](https://github.com/excid3/noticed/blob/797878fe65049151ce5375d596dd25ac96eec8d1/app/jobs/noticed/event_job.rb#L14) (and likely support just the same, as the same, for the bulk delivery job [kickoffs](https://github.com/excid3/noticed/blob/797878fe65049151ce5375d596dd25ac96eec8d1/app/jobs/noticed/event_job.rb#L8)). Does that seem reasonable?

**Comment by excid3** [comment]
`delay` is removed in favor of `wait` since they're all ActiveJobs and we don't need to introduce additional terminology.

That's all handled here:
https://github.com/excid3/noticed/blob/main/app/models/concerns/noticed/deliverable.rb#L23-L30

I also just added the ability to call `deliver(User.first, wait: 5.minutes)` in order to delay the EventJob after inserting the Event to the database. This would delay processing of the bulk and individual deliveries 5 minutes. Then each delivery method can have its own `wait` configuration separately as the EventJob simply enqueues the delivery jobs.

**Reply by jon-sully** [reply]
D'oh! Totally missed it. Thanks!

And nice 👍 that feels like a great add 💯
